components/Settings.jsx

```diff
@@ -93,12 +94,41 @@ export default Settings = ({ navigation }) => {
     }
   };
 
+  function handleSubmit() {
+    if (inputValid) {
+      if (newScreenName.length > 1) {
+        updateScreenName();
+      }
+      if (newFirstName.length > 1) {
+        updatedFirstName();
+      }
+      if (newLocation.length > 1) {
+        updatedLocation();
+      }
+      if (newLastName.length > 1) {
+        updatedLastName();
+      }
+    } else {
+      alert("Invalid entries please update the above fields.");
+    }
+  }
+
   return (
     <View style={styles.container}>
       <Text style={styles.header}>User Settings</Text>
       <View style={styles.inputGroup}>
         <Text>Change Username:</Text>
         <TextInput
+          inputMode="text"
+          textAlign="center"
+          onEndEditing={() => {
+            if (newScreenName.length < 5) {
```

Comment:
If I input a 2 letter word and click 'save' I get the message 'details successfully updated'. Not sure if you want to address this later or not